src/OpenBinary/reader.jl

```diff
@@ -156,137 +147,157 @@ function PSRI.open(
-            skip_store(ioh, 4, skips, store_skips)
+            skip(ioh, 4)
+            skip(ioh, 4)
+
             number_blocks = zeros(Int32, stage_total + 1 - first_relative_stage)
             offsets = zeros(Int32, 1 + stage_total + 1 - first_relative_stage)
-            for i in first_relative_stage:stage_total+1
-                offset = read(ioh, Int32)
-                offsets[i-first_relative_stage+1] = offset
-                if i > first_relative_stage
-                    number_blocks[i-first_relative_stage] = offset - offsets[i-first_relative_stage]
+            
+            for i in first_relative_stage:(stage_total+1)
+                let offset = read(ioh, Int32)
+                    offsets[i-first_relative_stage+1] = offset
+                    
+                    if i > first_relative_stage
+                        number_blocks[i-first_relative_stage] = offset - offsets[i-first_relative_stage]
+                    end
                 end
             end
+            
             block_total = maximum(number_blocks)
 
-            hour_discretization = floor(
-                Int,
-                block_total / if stage_type == PSRI.STAGE_WEEK
-                    168
-                elseif stage_type == PSRI.STAGE_MONTH
-                    if block_total % 672 == 0
-                        672
-                    elseif block_total % 720 == 0
-                        720
-                    else
-                        744
-                    end
-                elseif stage_type == PSRI.STAGE_DAY
-                    24
-                elseif stage_type == PSRI.STAGE_YEAR
-                    8760
-                else
-                    error("stage Type $stage_type not currently supported")
-                end,
-            )
+            hour_discretization = _get_hour_discretization(stage_type, block_total)
 
             if verbose_header
                 @show(block_total, length(number_blocks), number_blocks)
             end
         else
-            error("variable_by_hour = $variable_by_hour is invalid")
+            error("'variable_by_hour' must be either '0' or '1', not '$(variable_by_hour)'")
         end
     end
+
     agent_names = String[]
     agent_name_buffer = Vector{Cchar}(undef, name_length)
+
     for _ in 1:total_agents
-        skip_store(ioh, 4, skips, store_skips)
-        skip_store(ioh, 4, skips, store_skips)
+        skip(ioh, 4)
+        skip(ioh, 4)
+
         read!(ioh, agent_name_buffer)
         agent_name = strip(join(Char.(agent_name_buffer)))
         push!(agent_names, agent_name)
     end
+
     if verbose_header
         @show(agent_names)
     end
 
-    if single_binary
-        # nothing
-    else
+    if !single_binary
         close(ioh)
     end
 
-    BAD = "Bad header: "
+    if !(stage_total > 0)
+        error("'stage_total' must be a positive integer, not '$(stage_total)'")
+    end
 
-    @assert stage_total > 0
```

Comment:
We should at least check it this is larger or equal zero

Reply:
The error message that was added already indicates which value was given